devito/ops/node_factory.py

```diff
@@ -69,15 +69,10 @@ def new_ops_arg(self, indexed, is_write):
 
         return OpsAccess(symbol_to_access, space_indices)
 
-    def new_ops_gbl(self, c):
-        if c in self.ops_args:
-            return self.ops_args[c].accessible
+    def new_ops_gbl(self, expr):
+        if expr not in self.ops_args:
+            param = RawAccessible(name=expr.name, dtype=expr.dtype)
+            self.ops_args[expr] = AccessibleInfo(param, None, None, None)
+            self.ops_params.append(param)
 
-        new_c = AccessibleInfo(Constant(name='*%s' % c.name, dtype=c.dtype),
-                               None,
-                               None,
-                               None)
-        self.ops_args[c] = new_c
-        self.ops_params.append(new_c.accessible)
-
-        return new_c.accessible
+        return RawAccess(expr)
```

Comment:
So basically:
- change `self` state
- return another object that does not depend on it?

That feels odd

Reply:
i see what you mean.. 

this is one of the two base cases for the recursive method `make_ops_ast` (https://github.com/opesci/devito/blob/master/devito/ops/transformer.py#L296)

this case (`new_ops_gbl`) follows the same pattern from the other case (`new_ops_arg` - https://github.com/opesci/devito/blob/master/devito/ops/node_factory.py#L23)

Reply:
> change self state

@mloubout good spot ! 👏